scripts/zkai_attention_kv_risc0_scaled_sequence_receipt_gate.py

```
@@ -0,0 +1,959 @@
+
+def expected_input() -> dict[str, Any]:
+    fixture = sequence_fixture()
+    return {
+        "initial_kv_cache": fixture["initial_kv_cache"],
+        "input_steps": fixture["input_steps"],
+    }
+
+
+def dot(lhs: list[int], rhs: list[int]) -> int:
+    if len(lhs) != len(rhs):
+        raise AttentionKvRisc0SequenceReceiptError("dot-product width mismatch", layer="sequence_journal")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)